tests/test_filesystems.py

```diff
@@ -56,26 +78,49 @@ def test_must_contain_net_location(self):
         ):
             RemoteFileSystem(basepath="memory://")
 
+    def test_fails_basepath_is_file(self, tmp_path):
+        assert not tmp_path.is_file()
+
     async def test_read_write_roundtrip(self):
         fs = RemoteFileSystem(basepath="memory://root")
         await fs.write_path("test.txt", content=b"hello")
         assert await fs.read_path("test.txt") == b"hello"
 
+    def test_read_write_roundtrip_sync(self):
+        fs = RemoteFileSystem(basepath="memory://root")
+        fs.write_path("test.txt", content=b"hello")
+        assert fs.read_path("test.txt") == b"hello"
+
     async def test_write_with_missing_directory_succeeds(self):
         fs = RemoteFileSystem(basepath="memory://root/")
         await fs.write_path("memory://root/folder/test.txt", content=b"hello")
         assert await fs.read_path("folder/test.txt") == b"hello"
 
+    def test_write_with_missing_directory_succeeds_sync(self):
+        fs = RemoteFileSystem(basepath="memory://root/")
+        fs.write_path("memory://root/folder/test.txt", content=b"hello")
+        assert fs.read_path("folder/test.txt") == b"hello"
+
     async def test_write_outside_of_basepath_netloc(self):
         fs = RemoteFileSystem(basepath="memory://foo")
         with pytest.raises(ValueError, match="is outside of the base path"):
             await fs.write_path("memory://bar/test.txt", content=b"hello")
 
+    def test_write_outside_of_basepath_netloc_sync(self):
+        fs = RemoteFileSystem(basepath="memory://foo")
+        with pytest.raises(ValueError, match="is outside of the base path"):
+            fs.write_path("memory://bar/test.txt", content=b"hello")
```

Comment:
We actually don't want to cover every possible test case with `sync`. They go through the same code and duplicated coverage will be more work to maintain. We just want to make sure the method is usable, the `roundtrip` test should be all we need. Would you mind removing the extras?

Reply:
I removed all of those sync tests except the roundtrip sync ones